fuse/fuse.cpp

```diff
@@ -37,25 +38,78 @@
 #include "namespace.h"
 #endif
 
-int run_in_fuse(
-	const std::string& daemon_path,
-	const std::string& working_dir,
-	const std::string& json,
-	bool use_stdin_file,
-	std::string& result_json)
-{
-	if (0 != chdir(working_dir.c_str())) {
-		std::cerr << "chdir " << working_dir << ": " << strerror(errno) << std::endl;
-		return 1;
-	}
-
+bool json_as_struct(std::string json, json_args& result) {
 	JAST jast;
 	if (!JAST::parse(json, std::cerr, jast))
+		return false;
+
+	for (auto &x : jast.get("command").children)
+		result.command.push_back(x.second.value);
+
+	for (auto &x : jast.get("environment").children)
+		result.environment.push_back(x.second.value);
+
+	for (auto &x : jast.get("visible").children)
+		result.visible.push_back(x.second.value);
+
+	result.directory = jast.get("directory").value;
+	result.stdin_file = jast.get("stdin").value;
+
+#ifdef __linux__
+	result.isolate_network = jast.get("isolate-network").kind == JSON_TRUE;
+
+	std::string hostname = jast.get("hostname").value;
+	result.hostname = !hostname.empty() ? hostname : "build";
+
+	std::string domainname = jast.get("domainname").value;
+	result.domainname = !domainname.empty() ? domainname : "local";
```

Comment:
You probably need to leave this empty to detect when the domainname should be unmodified.